llmfoundry/callbacks/curriculum_learning_callback.py

```diff
@@ -193,25 +183,29 @@ def iteration_end(self, state: State, logger: Logger):
 
     def state_dict(self):
         return {
-            'schedule': self._schedule,
-            'schedule_index': self._schedule_index,
+            'state':
+                CurriculumLearningState(
```

Comment:
since this change breaks the backward compatibility, please make sure no existing checkpointing using this callback.

Reply:
Yes, I don't think anyone internally uses this